### PR TITLE
Updated game logic

### PR DESCRIPTION
**Pull request by matthewbeck**
1. Added a win and loss case

Upon win:
• target width is decreased by 1
• wait_update is cut in half

Upon loss:
• target width set back to 10
• wait_update set back to 100

2. Adjusted update_game so that we added "bounce effect" with counter increasing from 0, and then decreasing once reaching NUMLED

### CODE REVIEW

File: mainhub/mainhub.ino [hunk 1]
```diff
@@ -33,21 +33,26 @@ int cursor_index = 0;
...
   Serial.println("Starting ArcGame...");
+  delay(1000);
 
   pinMode(PIN_LED, OUTPUT);
   digitalWrite(PIN_LED, HIGH); // Turn the built-in led on to show there is power
 
   // Additional Setup steps go here
 
   Serial.println("Init Complete...");
+  delay(1000);
```
Comment (jazzz) [inline mainhub/mainhub.ino:55]: Are these Delays helping?

Usually only the first one is needed.

File: mainhub/mainhub.ino [hunk 2]
```diff
@@ -78,18 +83,29 @@ void draw()
 
 void update_game()
 {
-
+  cursor_index += direction; // Move to the next LED based on the direction
+
+  // Check if the LED reached the end of the strip
+  if (cursor_index >= NUM_LEDS) {
+    cursor_index = NUM_LEDS - 2; // Move to the second-to-last LED
```
Comment (jazzz) [inline mainhub/mainhub.ino:90]: Great work avoiding the 'off-by-one' error!

File: mainhub/mainhub.ino [hunk 3]
```diff
@@ -78,18 +83,29 @@ void draw()
...
+  }
   // Increment the cursor, but reset to zero if it is greater than the number of leds (Circle Path)
-  cursor_index = cursor_index + 1;
+  /*cursor_index = cursor_index + 1;
   if (cursor_index >= NUM_LEDS)
   {
     cursor_index = 0;
   }
-
+*/
```
Comment (jazzz) [inline mainhub/mainhub.ino:102]: Doesn't look like you need this anymore. Its scary to delete code, but its saved on git so you can always come back to it : https://github.com/LumenFactory/arc-game/commits/main

File: mainhub/mainhub.ino [hunk 4]
```diff
@@ -114,6 +130,28 @@ void loop()
...
-
-  //  delay(10);
+  if (is_button_pressed) {
+    if (cursor_index >= min_index && cursor_index <= max_index) {
+        // Game won!
+      Serial.println("Game won!");
+      delay(1000);
+      cursor_index = 0;
+      target_width -= 1;
+      wait_update = wait_update/2;
```
Comment (jazzz) [inline mainhub/mainhub.ino:140]: This looks great!

For managing future complexity, it might be helpful to have a variable `int current_level` which tracks which level you are on. 

that way you can derive all the other values such as the animations and speeds from it.

**Review by jazzz — APPROVED** [review]
@matthewbeck This is looking so good. 

I like how clean it is so far, and your use of comments. Very easy to follow.

**Comment by jazzz** [comment]
In the future, it might be helpful to work on your own branch as it helps keep everything organized. Doubt there will be any issues here, but never hurts to get into the habit. 

Creating a new branch is a lightweight operation and is never the wrong thing to do. 


This is looking great. Cleanup anything you feel the need to clean up, then merge it into main to make it official. Merging to main often is best as big changes get hard to follow.

When you are ready for your next branch make sure you build it off of main to get the latest code.

Looking so good so far!

**Comment by matthewbeck** [comment]
@jazzz should I clean up the code locally on my computer and then submit a new PR or just do it in the branch?

**Comment by jazzz** [comment]
> should I clean up the code locally on my computer and then submit a new PR or just do it in the branch?

On the Branch. Think of it as 1 branch: 1Pr: Many commits . 

One thing I like to do sometimes is to open a draft PR as soon as I start working, as a way of tracking what I am doing and sharing it with others. Then once I'm happy with it it gets pushed into main
